packages/chain-adapters/src/evm/EvmBaseAdapter.ts

```diff
@@ -187,7 +187,7 @@ export abstract class EvmBaseAdapter<T extends EvmChainId> implements IChainAdap
           tx.value = bnOrZero(account.balance).minus(fee).toString()
         }
       }
-      const data = await getErc20Data(to, tx.value, erc20ContractAddress)
+      const data = tx.memo || (await getErc20Data(to, tx.value, erc20ContractAddress))
```

Comment:
Safer than `??` WRT nullish strings, memo should be undefined or a non-empty string but you never know what consumers might pass